src/gui/shader.cpp

```diff
@@ -29,82 +34,99 @@ load_shaders(
+    } else {
+        LOG_ERROR(
+            logger, "Cannot open {}. Are you in the right directory?",
+            fragment_file_path
+        );
+        return 0;
     }
 
     GLint Result = GL_FALSE;
     int info_log_length;
 
     // Compile Vertex Shader
-    std::cout << "Compiling shader : " << vertex_file_path << std::endl;
+    LOG_INFO(logger, "Compiling vertex shader {}", vertex_file_path);
+
     char const* vertex_source_pointer = vertex_shader_code.c_str();
-    glShaderSource(vertex_shader_ID, 1, &vertex_source_pointer, NULL);
-    glCompileShader(vertex_shader_ID);
+    glShaderSource(vertex_shader_id, 1, &vertex_source_pointer, NULL);
+    glCompileShader(vertex_shader_id);
 
     // Check Vertex Shader
-    glGetShaderiv(vertex_shader_ID, GL_COMPILE_STATUS, &Result);
-    glGetShaderiv(vertex_shader_ID, GL_INFO_LOG_LENGTH, &info_log_length);
+    glGetShaderiv(vertex_shader_id, GL_COMPILE_STATUS, &Result);
+    glGetShaderiv(vertex_shader_id, GL_INFO_LOG_LENGTH, &info_log_length);
     if (info_log_length > 0) {
-        std::vector<char> vertex_shader_error_message(info_log_length + 1);
+        std::string vertex_shader_error_message(info_log_length + 1, '\0');
         glGetShaderInfoLog(
-            vertex_shader_ID, info_log_length, NULL, &vertex_shader_error_message[0]
+            vertex_shader_id, info_log_length, NULL, vertex_shader_error_message.data()
         );
-        printf("%s\n", &vertex_shader_error_message[0]);
+
+        LOG_WARNING(logger, "Vertex shader error: {}", vertex_shader_error_message);
     }
 
     // Compile Fragment Shader
-    std::cout << "Compiling shader : " << fragment_file_path << std::endl;
+    LOG_INFO(logger, "Compiling fragment shader {}", fragment_file_path);
+
     char const* fragment_source_pointer = fragment_shader_code.c_str();
-    glShaderSource(fragment_shader_ID, 1, &fragment_source_pointer, NULL);
-    glCompileShader(fragment_shader_ID);
+    glShaderSource(fragment_shader_id, 1, &fragment_source_pointer, NULL);
+    glCompileShader(fragment_shader_id);
 
     // Check Fragment Shader
-    glGetShaderiv(fragment_shader_ID, GL_COMPILE_STATUS, &Result);
-    glGetShaderiv(fragment_shader_ID, GL_INFO_LOG_LENGTH, &info_log_length);
+    glGetShaderiv(fragment_shader_id, GL_COMPILE_STATUS, &Result);
+    glGetShaderiv(fragment_shader_id, GL_INFO_LOG_LENGTH, &info_log_length);
     if (info_log_length > 0) {
-        std::vector<char> fragment_shader_error_message(info_log_length + 1);
+        std::string fragment_shader_error_message(info_log_length + 1, '\0');
         glGetShaderInfoLog(
-            fragment_shader_ID, info_log_length, NULL, &fragment_shader_error_message[0]
+            fragment_shader_id, info_log_length, NULL,
+            fragment_shader_error_message.data()
         );
-        printf("%s\n", &fragment_shader_error_message[0]);
+
+        LOG_WARNING(logger, "Vertex shader error: {}", fragment_shader_error_message);
```

Comment:
fragment_shader_error_message is a vector. Is there a better way to log this?

Reply:
No its not I fixed your UB

![image](https://user-images.githubusercontent.com/76668284/208224369-223d1fc9-bdb2-4301-b3b9-34581f3d8512.png)

Reply:
ok sure, but it's multi line, and adds a blank line to the log

Reply:
How does it add a blank line?

Reply:
1) cause an error in a vertex shader
2) read the log file.